### PR TITLE
Issuer URI has been passed through variable

### PR DESCRIPTION
**Pull request by abalasub1312**
As per issue #59 we have added issuer_uri as configurable variable. Along with that I have added validation to force user to enter https:// as URI and not http://.

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@abalasub1312
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/modules/gh-oidc/README.md /tmp/tmp.6qdSXQWazy/generate_docs/workspace/modules/gh-oidc/README.md
73a74
> | issuer\_uri | Workload Identity Pool Issuer URL | `string` | n/a | yes |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
modules/gh-oidc/variables.tf
--- old/modules/gh-oidc/variables.tf
+++ new/modules/gh-oidc/variables.tf
@@ -46,11 +46,11 @@
   description = "Workload Identity Pool Issuer URL"
   validation {
     condition = (
-        length(var.issuer_uri) > 8 &&
-        substr(var.issuer_uri, 0, 8) == "https://"
+      length(var.issuer_uri) > 8 &&
+      substr(var.issuer_uri, 0, 8) == "https://"
     )
     error_message = "The Issuer URL value must start with \"https://\"."
-    }
+  }
 }
 
 variable "provider_display_name" {
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

File: modules/gh-oidc/variables.tf [hunk 1]
```diff
@@ -41,6 +41,18 @@ variable "provider_id" {
   description = "Workload Identity Pool Provider id"
 }
 
+variable "issuer_uri" {
```
Comment (bharathkkb) [inline modules/gh-oidc/variables.tf:44]: Lets default to "https://token.actions.githubusercontent.com"

File: modules/gh-oidc/variables.tf [hunk 2]
```diff
@@ -41,6 +41,18 @@ variable "provider_id" {
...
+  type        = string
+  description = "Workload Identity Pool Issuer URL"
+  validation {
+    condition = (
+        length(var.issuer_uri) > 8 &&
+        substr(var.issuer_uri, 0, 8) == "https://"
+    )
+    error_message = "The Issuer URL value must start with \"https://\"."
+    }
```
Comment (bharathkkb) [inline modules/gh-oidc/variables.tf:53]: We try to minimize module specific validation., we can skip this.

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR @abalasub1312 
Please also run `make docker_generate_docs` and `terraform fmt -recursive` per bot suggestion in https://github.com/terraform-google-modules/terraform-google-github-actions-runners/pull/62#issuecomment-1168528148

**Comment by comment-bot-dev** [comment]
@abalasub1312
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**